### PR TITLE
Factor PR validation Yaml files

### PR DESCRIPTION
**Pull request by dannyvv**
Factor out PR validation yaml files in preparation for finer grained control and unifying PR and CI steps

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6635)

### CODE REVIEW

**Review by JunielKatarn — APPROVED** [review]

File: .ado/jobs/playground.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          TestWACK: true
+        X86Debug_Win32:
+          BuildConfiguration: Debug
+          BuildPlatform: x86
+          SolutionFile: Playground-Win32.sln
+        X64Release_Win32:
+          BuildConfiguration: Release
+          BuildPlatform: x64
+          SolutionFile: Playground-Win32.sln
+        X86Debug_WinUI:
```
Comment (NickGerleman) [inline .ado/jobs/playground.yml:38]:
```suggestion
        X86Debug_WinUI3:
```

**Review by NickGerleman — APPROVED** [review]

**Comment by NickGerleman** [comment]
General nit, would be nice to standardize between spaces vs underscores vs camelcase for separating flavors. Right now it seems like we have a mix of different casing or separators.

**Comment by NickGerleman** [comment]
Looks like WinUI3 init tests continueOnError might not be being propagated correctly https://dev.azure.com/ms/react-native-windows/_build/results?buildId=125270&view=logs&j=f4eb2386-2984-5ed8-ca76-8a83d52bcb4d&t=0285b28e-6a0d-52e2-da6a-c5f865da8290&l=5535

I remember fighting super hard with having that work correctly in the matrix...

**Comment by dannyvv** [comment]
> General nit, would be nice to standardize between spaces vs underscores vs camelcase for separating flavors. Right now it seems like we have a mix of different casing or separators.

Make  a call and I'll make it happen :)

File: .ado/jobs/playground.yml (hunk 1 above)
Comment (dannyvv) [reply .ado/jobs/playground.yml]: Changed it and removed underscore.

**Comment by dannyvv** [comment]
> Looks like WinUI3 init tests continueOnError might not be being propagated correctly https://dev.azure.com/ms/react-native-windows/_build/results?buildId=125270&view=logs&j=f4eb2386-2984-5ed8-ca76-8a83d52bcb4d&t=0285b28e-6a0d-52e2-da6a-c5f865da8290&l=5535
> 
> I remember fighting super hard with having that work correctly in the matrix...

@NickGerleman: Thanks for pointing this out and your help with your beachball insight this evening with trying to get a repro. This is indeed an azure devops limmitation and I tried many expression types and it needs to be a const. continueOnError does not allow expressions in ado... I had to keep the 3 seperate CLi init cases :(

**Pull request merged**